winch/codegen/src/isa/aarch64/asm.rs

```diff
@@ -901,4 +908,32 @@ impl Assembler {
     pub fn buffer(&self) -> &MachBuffer<Inst> {
         &self.buffer
     }
+
+    pub fn call_with_name(&mut self, name: UserExternalNameRef) {
+        self.emit(Inst::Call {
+            info: Box::new(cranelift_codegen::CallInfo::empty(
+                ExternalName::user(name),
+                CallConv::SystemV,
+            )),
+        })
+    }
+
+    pub fn call_with_reg(&mut self, callee: Reg) {
+        self.emit(Inst::CallInd {
+            info: Box::new(cranelift_codegen::CallInfo::empty(
+                callee.into(),
+                CallConv::SystemV,
```

Comment:
> Right now the calling convention is hardcoded to SystemV, but maybe we also want to support AppleAarch64?

I see why this might be confusing. Even though the `CallConv` is hard coded here, there are two things to note:

* All the infrastructure to handle the calling convention differences, is handled very early on in the compilation process in Winch (and in Cranelift as well) e.g., when constructing the signature for the callee, at that point,[ you'll note that Winch correctly handles the default and apple-aarch64 calling conventions.](https://github.com/bytecodealliance/wasmtime/blob/main/winch/codegen/src/isa/aarch64/abi.rs#L33)
* Winch relies on Cranelift's binary emission infrastructure, and in this concrete case, as you can note, you're constructing a `CallInfo::empty()`, which signals that none of the other Cranelift-specific invariants must be taken into account when emitting calls (some of them depend on the calling convention that is passed here). 

I've opened https://github.com/bytecodealliance/wasmtime/pull/9757 to refactor call emission a bit with the hope of making this part more consistent.

Reply:
That refactor makes a lot of sense to me 👍

Reply:
updated in f4a6cf7